provider/resource_corp_list.go

```diff
@@ -22,17 +22,35 @@ func resourceCorpList() *schema.Resource {
 				Description: "Descriptive list name",
 				Required:    true,
 				ForceNew:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if !validStringLength(val.(string), 3, 32) {
+						return nil, []error{fmt.Errorf(`received name %q is invalid. should be min len 3, max len 32`, val.(string))}
+					}
+					return nil, nil
+				},
 			},
 			"type": {
 				Type:        schema.TypeString,
 				Description: "List types (string, ip, country, wildcard, signal)",
 				Required:    true,
 				ForceNew:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if !existsInString(val.(string), "string", "ip", "country", "wildcard", "signal") {
+						return nil, []error{fmt.Errorf(`received type %q is invalid. should be "string", "ip", "country", "wildcard" or "signal"`, val.(string))}
+					}
+					return nil, nil
+				},
 			},
 			"description": {
 				Type:        schema.TypeString,
 				Description: "Optional list description",
 				Optional:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if !validStringLength(val.(string), 0, 140) {
+						return nil, []error{fmt.Errorf(`received name %q is invalid. should be max len 140`, val.(string))}
```

Comment:
```suggestion
						return nil, []error{fmt.Errorf(`received description %q is invalid. should be max len 140`, val.(string))}
```

Reply:
oh sorry! thank you for your suggestion.